app/services/scraper/bedelias.rb

```diff
@@ -0,0 +1,141 @@
+require 'capybara'
+require 'capybara/dsl'
+
+module Scraper::Bedelias
+  extend self
+
+  include Capybara::DSL
+
+  # 5265 - CIENCIAS HUMANAS Y SOCIALES - min: 10 créditos
+  # SRN14 - MATEMÁTICA DISCRETA I - créditos: 10
+  CODE_NAME_CREDITS_REGEX = /\A(\w+) - (.+) - (?:min|créditos): (\d+)( créditos)?\z/
+  MAX_PAGES = ENV["MAX_PAGES"]&.to_i
+
+  def scrape
+    Capybara.configure do |config|
+      config.default_driver = ENV["HEADLESS"] == "false" ? :selenium_chrome : :selenium_chrome_headless
+      config.run_server = false
+      config.save_path = "tmp"
+    end
+
+    groups = {}
+    subjects = {}
+
+    go_to_groups_and_subjects_page
+    process_groups_and_subjects(groups, subjects)
+
+    go_to_prerequisites_page
+    prerequisites = process_prerequisites(subjects)
+
+    prerequisites.each do |prerequisite_tree|
+      add_missing_exams_and_subjects(prerequisite_tree, subjects)
+    end
+
+    File.write(Rails.root.join("db/data/scraped_subject_groups.yml"), groups.deep_stringify_keys.to_yaml)
+    File.write(Rails.root.join("db/data/scraped_subjects.yml"), subjects.deep_stringify_keys.to_yaml)
+    File.write(Rails.root.join("db/data/scraped_prerequisites.yml"), { prerequisites: }.deep_stringify_keys.to_yaml)
+  rescue
+    Rails.logger.info save_screenshot
+    raise
+  end
+
+  private
+
+  def go_to_groups_and_subjects_page
+    visit "https://bedelias.udelar.edu.uy"
+    click_on "PLANES DE ESTUDIO"
+    click_on "Planes de estudio / Previas"
+
+    execute_script("$.fx.off = true;") # Disable jQuery effects
+
+    find('h3', text: 'TECNOLOGÍA Y CIENCIAS DE LA NATURALEZA').click
+    find('span', text: 'FING - FACULTAD DE INGENIERÍA', visible: false).click
+
+    within('tr', text: 'INGENIERIA EN COMPUTACION', match: :prefer_exact) do
+      find('.ui-row-toggler').click
+    end
+    within('.ui-expanded-row-content', text: 'Planes') do
+      find('tr', text: '1997').click_on "Ver más datos"
+    end
+  end
+
+  def process_groups_and_subjects(groups, subjects)
+    node_type = "[data-nodetype='Grupo']"
+    selector = ".arbolComposicion #{node_type}:not(:has(#{node_type})) > .ui-treenode-content > .ui-treenode-label"
+
+    all(selector, visible: false).each do |node|
```

Comment:
i couldn't make it exactly like this because the line would be longer to 120 chars, but i kept the selector var, and defined `group_nodes` in 8831a8edcb8864cb3ae6641c98f62066addd8873 like you did. Let me know what you think

Reply:
LGTM. Thanks!